lms/djangoapps/grades/tests/test_new.py

```diff
@@ -230,3 +235,168 @@ def test_save_and_load(self):
 
         self.assertEqual(input_grade.url_name, loaded_grade.url_name)
         self.assertEqual(input_grade.all_total, loaded_grade.all_total)
+
+
+@ddt.ddt
+class TestMultipleProblemTypesSubsectionScores(ModuleStoreTestCase, ProblemSubmissionTestMixin):
+    """
+    Test grading of different problem types.
+    """
+
+    default_problem_metadata = {
+        u'graded': True,
+        u'weight': 2.5,
+        u'max_score': 7.0,
+        u'due': datetime.datetime(2099, 3, 15, 12, 30, 0, tzinfo=pytz.utc),
+    }
+
+    COURSE_NAME = u'Problem Type Test Course'
+    COURSE_NUM = u'probtype'
+
+    def setUp(self):
+        super(TestMultipleProblemTypesSubsectionScores, self).setUp()
+        password = u'test'
+        self.student = UserFactory.create(is_staff=False, username=u'test_student', password=password)
+        self.client.login(username=self.student.username, password=password)
+        self.request = get_request_for_user(self.student)
+        self.course = CourseFactory.create(
+            display_name=self.COURSE_NAME,
+            number=self.COURSE_NUM
+        )
+        self.chapter = ItemFactory.create(
+            parent=self.course,
+            category=u'chapter',
+            display_name=u'Test Chapter'
+        )
+        self.seq1 = ItemFactory.create(
+            parent=self.chapter,
+            category=u'sequential',
+            display_name=u'Test Sequential 1',
+            graded=True
+        )
+        self.vert1 = ItemFactory.create(
+            parent=self.seq1,
+            category=u'vertical',
+            display_name=u'Test Vertical 1'
+        )
+
+    def _add_block_from_xml_file(self, block_type, filename, parent, metadata=None):
```

Comment:
This seems like a generally useful function.  Can we move it somewhere in core/openedx?